x/dsmr/certificate.go

```diff
@@ -39,35 +39,58 @@ func init() {
 	}
 }
 
-type NoVerifyChunkSignatureShare struct{}
-
-func (NoVerifyChunkSignatureShare) Verify(_ ids.ID) error { return nil }
-
-type NoVerifyChunkSignature struct{}
-
-func (NoVerifyChunkSignature) Verify() error { return nil }
-
-type ChunkCertificate struct {
+type ChunkCertificate[T Tx] struct {
 	ChunkID ids.ID `serialize:"true"`
 	Expiry  int64  `serialize:"true"`
 
-	Signature NoVerifyChunkSignature `serialize:"true"`
+	Signature *warp.BitSetSignature `serialize:"true"`
 }
 
-func (c *ChunkCertificate) GetChunkID() ids.ID { return c.ChunkID }
+func (c *ChunkCertificate[_]) GetChunkID() ids.ID { return c.ChunkID }
 
-func (c *ChunkCertificate) GetSlot() int64 { return c.Expiry }
+func (c *ChunkCertificate[_]) GetSlot() int64 { return c.Expiry }
 
-func (c *ChunkCertificate) Bytes() []byte {
+func (c *ChunkCertificate[_]) Bytes() []byte {
 	bytes, err := Codec.Marshal(CodecVersion, c)
 	if err != nil {
 		panic(err)
 	}
 	return bytes
 }
 
-func (c *ChunkCertificate) Verify(_ context.Context, _ interface{}) error {
-	return c.Signature.Verify()
+func (c *ChunkCertificate[T]) Verify(
+	ctx context.Context,
+	storage *ChunkStorage[T],
+	networkID uint32,
+	chainID ids.ID,
+	pChainState validators.State,
+	pChainHeight uint64,
+	quorumNum uint64,
+	quorumDen uint64,
+) error {
+	chunkBytes, _, err := storage.GetChunkBytes(c.Expiry, c.ChunkID)
```

Comment:
We should remove this check, we don't need the chunk available in local storage for it to be valid for inclusion within a block